notebooks/02_model/hypertune_spark_deep_dive.ipynb

```diff
@@ -0,0 +1,983 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Hyperparameter tuning (Spark based recommender)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Hyperparameter tuning for Spark based recommender algorithm is important to select a model with the optimal performance. This notebook introduces good practices in performing hyperparameter tuning for building recommender models with the utility functions provided in the [Microsoft/Recommenders](https://github.com/Microsoft/Recommenders.git) repository."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 0 Global settings and import"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "System version: 3.6.0 | packaged by conda-forge | (default, Feb  9 2017, 14:36:55) \n",
+      "[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]\n",
+      "Pandas version: 0.23.4\n",
+      "PySpark version: 2.3.1\n"
+     ]
+    }
+   ],
+   "source": [
+    "# set the environment path to find Recommenders\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
+    "import sys\n",
+    "sys.path.append(\"../../\")\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import time\n",
+    "\n",
+    "import pyspark\n",
+    "from pyspark import SparkContext, SparkConf\n",
+    "from pyspark.ml.recommendation import ALS\n",
+    "from pyspark.ml.tuning import ParamGridBuilder, TrainValidationSplit\n",
+    "from pyspark.ml.evaluation import RegressionEvaluator\n",
+    "\n",
+    "from hyperopt import fmin, tpe, hp, STATUS_OK, Trials\n",
```

Comment:
Do we need to add this hypteropt to the conda file?

Reply:
Removed.